vendor/bat-native-ledger/src/bat/ledger/internal/publisher/publisher_list_reader.cc

```diff
@@ -0,0 +1,81 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ledger/internal/publisher/publisher_list_reader.h"
+
+#include <utility>
+
+#include "bat/ledger/internal/common/brotli_helpers.h"
+#include "bat/ledger/internal/publisher/prefix_util.h"
+
+namespace braveledger_publisher {
+
+PublisherListReader::PublisherListReader()
+    : prefix_size_(kMinPrefixSize) {}
+
+PublisherListReader::~PublisherListReader() = default;
+
+PublisherListReader::ParseError PublisherListReader::Parse(
+    const std::string& contents) {
+  using publishers_pb::PublisherList;
+
+  PublisherList message;
+  if (!message.ParseFromString(contents)) {
+    return ParseError::InvalidProtobufMessage;
+  }
+
+  size_t prefix_size = message.prefix_size();
+  if (prefix_size < kMinPrefixSize || prefix_size > kMaxPrefixSize) {
+    return ParseError::InvalidPrefixSize;
+  }
+
+  size_t uncompressed_size = message.uncompressed_size();
+  if (uncompressed_size == 0 || uncompressed_size % prefix_size != 0) {
+    return ParseError::InvalidUncompressedSize;
+  }
+
+  std::string uncompressed;
+  switch (message.compression_type()) {
```

Comment:
why are we compressing data inside the publisher list instead of just compressing the entire response payload?

Reply:
This design is inspired by Safe Browsing, and will give us a lot of flexibility for different compression approaches in the future (similar to how Safe Browsing uses Rice encoding).